protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -105,6 +117,131 @@ struct ScarbStarknetContractArtifact {
     casm: Option<PathBuf>,
 }
 
+fn execute_syscall(
+    system: &ResOperand,
+    vm: &mut VirtualMachine,
+    blockifier_state: &mut CachedState<DictStateReader>,
+) -> Result<(), HintError> {
+    let (cell, offset) = extract_buffer(system);
+    let mut system_ptr = get_ptr(vm, cell, &offset)?;
+
+    let selector = vm.get_integer(system_ptr).unwrap();
+    // dbg!(&selector);
+    let selector = selector.to_bytes_be();
+    system_ptr += 1;
+
+    let gas_counter = vm.get_integer(system_ptr).unwrap().to_usize().unwrap();
+    system_ptr += 1;
+
+    // Get
+    // let x = vm.get_integer(system_ptr).unwrap().to_owned();
+    let contract_address = vm.get_integer(system_ptr).unwrap().clone().into_owned();
+    system_ptr += 1;
+
+    let entry_point_selector = vm.get_integer(system_ptr).unwrap().into_owned();
+    system_ptr += 1;
+
+    let mut start = vm.get_relocatable(system_ptr).unwrap();
+    system_ptr += 1;
+
+    let end = vm.get_relocatable(system_ptr).unwrap();
+    system_ptr += 1;
+
+    let mut calldata = vec![];
+    while start != end {
+        let val = vm.get_integer(start).unwrap().into_owned();
+        calldata.push(val);
+        start.offset += 1;
+    }
```

Comment:
```suggestion
    let mut calldata = vm_get_range(vm, start, end)?;
```

Reply:
It's not a public method in cairo compiler.